travis/test_pylint

```diff
@@ -1,23 +1,188 @@
+    path = os.getenv('PATH')
+    for item in path.split(os.path.pathsep):
+        item = os.path.join(item, name)
+        if os.path.exists(item):
+            return item
+
+
+def main():
+    '''
+    Method to run pylint.
+    :return: Boolean if all fine return True else False
+    '''
+    conf_file = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        'cfg',
+        "travis_run_pylint.cfg")
+    paths_to_check = [
+        os.path.join(os.environ['TRAVIS_BUILD_DIR'], module)
+        for module in get_modules(
+            os.environ['TRAVIS_BUILD_DIR'])
+    ]
+    odoo_path = get_default_travis_server_path()
+    addons_path = get_default_travis_addons_path()
+    instance_symlink_path = None
+    sys_paths = []
+    version = os.environ.get('VERSION', 'master')
+    extra_params = get_extra_params_from_version(version)
+    odoo_lint_path = which('pylint_plugins')
+    if odoo_lint_path and os.path.exists(os.path.join(
+       odoo_lint_path, 'odoo_lint.py')):
+        extra_params.append('--load-plugins=odoo_lint')
+        sys_paths.append(odoo_lint_path)
+
+    if os.path.exists(odoo_path):
+        instance_symlink_path = os.path.join(
+            tempfile.gettempdir(), 'instance')
+        setup_server_symlink(odoo_path, addons_path,
+                             instance_symlink_path)
+    if instance_symlink_path:
+        sys_paths.append(instance_symlink_path)
+
+    result = True
+    linter_stats = run_pylint(
+        paths_to_check, conf_file, sys_paths,
+        extra_params=extra_params)
+    number_errors = get_number_errors(linter_stats)
+    is_pull_request = os.environ.get('TRAVIS_PULL_REQUEST', 'false') != 'false'
+    if is_pull_request:
+        # If is a pr then use new config file of pylint
+        future_conf_file = os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            'cfg',
+            "travis_run_pylint_pr.cfg")
+        branch_name_diff = os.environ.get('TRAVIS_BRANCH', False)
+        if branch_name_diff:
+            modules_changed = get_modules_changed(
+                os.environ['TRAVIS_BUILD_DIR'],
+                'origin/' + branch_name_diff)
+            paths_module_changed = [
+                os.path.join(os.environ['TRAVIS_BUILD_DIR'], module_changed)
+                for module_changed in modules_changed]
+            if paths_module_changed:
+                linter_stats = run_pylint(
+                    paths_module_changed, future_conf_file,
+                    sys_paths, extra_params=extra_params)
+                number_errors += get_number_errors(linter_stats)
+    expected_errors = int(
+        os.environ.get("PYLINT_EXPECTED_ERRORS", "0"))
+    if number_errors != expected_errors:
+        print("pylint expected errors {expected_errors}, "
+              "found {number_errors}!".format(
+                  expected_errors=expected_errors,
+                  number_errors=number_errors))
+        result = False
```

Comment:
I used `PYLINT_EXPECTED_ERRORS` in `test_pylint` script because we have to consider git hook script.
Maybe, I have a local error expected and I need to defined in commit sentence.
`PYLINT_EXPECTED_ERROR=1 git commit -m "my msg"`
What do you think?